fe/lib/components/BrandSelect/BrandSelect.tsx

```diff
@@ -0,0 +1,119 @@
+import { ApolloClient, useQuery } from '@apollo/client';
+import {
+  FieldMessage,
+  Heading,
+  Inline,
+  Loader,
+  Stack,
+  Text,
+} from 'braid-design-system';
+import React, { useEffect, useState } from 'react';
+
+import {
+  AdvertisementBrandingFieldsFragment,
+  AdvertisementBrandingsQuery,
+  AdvertisementBrandingsQueryVariables,
+} from '../../types/seekApi.graphql';
+
+import { PaginatedBrands } from './PaginatedBrands';
+import { ADVERTISEMENT_BRANDINGS } from './queries';
+
+export interface BrandSelectProps {
+  client?: ApolloClient<unknown>;
+  hirerId: string;
+  errorMessage?: string;
+  initialBrandId?: string;
+  pageSize?: number;
+  /**
+   * Type definition from the `Controller` component that's provided by `react-hook-form`.
+   */
+  onChange: (...event: any[]) => void;
```

Comment:
1. What's going on with the type here? This is pretty unusable for consumers.
1. All the other components seem to name this `onSelect`; we should align these one way or the other.